src/Microsoft.Health.Dicom.Core/Extensions/DicomDatasetExtensions.cs

```diff
@@ -329,5 +329,35 @@ public static void ValidateQueryTag(this DicomDataset dataset, QueryTag queryTag
                 minimumValidator.Validate(dicomElement);
             }
         }
+
+        /// <summary>
+        /// Gets DicomDatasets that matches a DicomSequence.
+        /// </summary>
+        /// <param name="dataset">The DicomDataset to be traversed.</param>
+        /// <param name="searchSequence">The Dicom sequence modelling the path.</param>
+        /// <returns>Lists of DicomDataset that matches the search sequence.</returns>
+        public static IEnumerable<DicomDataset> SearchSequencePath(this DicomDataset dataset, DicomSequence searchSequence)
```

Comment:
I am trying to understand this API. Are you trying to find an array of values that match the TagPath?

If that is the case, is there a opportunity to simplify the input and return type?

Input can be TagPath and output can be List of values based on the VR type

Reply:
Also to be consistent call the Method GetSequencePathValues<T>()

Reply:
This method returns the matching elements inside a dicom sequence elements, the original dataset can contain all the items but the search sequence will contain only the indexed element that we need. And the returned values can be list of elements but for our logic further we need to know the dataset so we can use the existing code.

Reply:
Updated to GetSequencePathValues and instead of sequence we now pass list of dicom tags.